src/core/node/node.builder.js

```diff
@@ -44,7 +45,7 @@ export default function nodeBuilder(columnMap, groupBy, valueFactory, level = 0)
 			for (let i = 0, length = keys.length; i < length; i++) {
 				const key = keys[i];
 				const group = groups[key];
-				group.node.children = build(group.rows);
+				group.node.children = build(group.rows, group.node.rows);
```

Comment:
For style i'd love more if pass delegate to avoid checking rowIndexes ? ...:
```javascript
build(group.rows, i => group.node.rows[i] || identity)
```
I'm not sure about perfomace